resolwe_bio/processes/expression/featureCounts.yml

```diff
@@ -153,54 +388,115 @@
     runtime: polyglot
     language: bash
     program: |
-      NAME=`basename {{ alignments.bam.file}} .bam`
+      {% if not alignment.reads.species == annotation.annotation.species %}
+        re-error 'Species of aligned reads and annotation do not match.'
+      {% endif %}
 
-      COUNT_MULTI_MAPPING_READS="{% if count_multi_mapping_reads %} -M{% endif %}"
+      {% if not alignment.reads.build == annotation.annotation.build %}
+        re-error 'Build of aligned reads and annotation do not match.'
+      {% endif %}
 
-      IS_PAIRED_END="{% if PE_options.is_paired_end %} -p{% endif %}"
-      REQUIRE_BOTH_ENDS_MAPPED="{% if PE_options.require_both_ends_mapped %} -B{% endif %}"
-      COUNT_CHIMERIC_FRAGMENTS="{% if PE_options.count_chimeric_fragments %} -C{% endif %}"
+      {% set alignment_fn = alignment.reads.bam.file|basename %}
+      {% set name = alignment_fn[:-4]
+         if alignment_fn.endswith('.bam')
+         else alignment_fn %}
+      {% set annotation_fmt = 'gtf'
+         if annotation.annotation|type|subtype('data:annotation:gtf')
+         else 'gff3' %}
+      {% set strandedness = {'non_specific': 0, 'forward': 1, 'reverse': 2} %}
 
-      # Check if reads are paired or single-ended.
-      if [[ $(samtools view -c -f 1 {{ alignments.bam.file }}) != 0 ]]; then
+      if [[ $(samtools view -c -f 1 {{ alignment.reads.bam.file }}) == 0 ]]
+      then
         featureCounts \
-          -a {{annotation.annot.file}} \
-          -o "featureCounts_rc.txt" \
-          -g {{ id_attribute }} \
-          ${COUNT_MULTI_MAPPING_READS} \
-          -Q {{ min_mqs }} \
-          -s {{ strand_specific }} \
-          -t {{ feature_class }} \
-          ${REQUIRE_BOTH_ENDS_MAPPED} \
-          ${COUNT_CHIMERIC_FRAGMENTS} \
-          ${IS_PAIRED_END} \
-          -R {{ alignments.bam.file }}
+          -a {{ annotation.annotation.annot.file }} \
+          -o featureCounts_rc.txt \
+          -F GTF \
+          -t {{ annotation.feature_class }} \
+          -g {{ annotation.id_attribute }} \
+          {% if advanced.summarization_level.count_features %} -f {% endif %} \
+          {% if advanced.summarization_level.count_features and advanced.overlap.allow_multi_overlap %} \
+            -O \
```

Comment:
can user specify only allow_multi_overlap? Add to description and warning message that count_features is needed for allow_multi_overlap feature.

Reply:
Good catch! This is a mistake. `allow_multi_overlap` should be available regardless of the `count_features` choice.